python/ray/data/datasource/sql_datasource.py

```diff
@@ -82,12 +71,17 @@ def _connect(connection_factory: Callable[[], Connection]) -> Iterator[Cursor]:
         connection.close()
 
 
-class _SQLReader(Reader):
+@DeveloperAPI
```

Comment:
Let's keep it private and not add `DeveloperAPI`. I feel it's too early. Let's gather feedback for our `read_sql` in 2.4 release first.